experiments/control/agents/QLearning.py

```diff
@@ -1,72 +1,38 @@
-import numpy as np
 import torch
 import torch.nn.functional as f
-import torch.optim as optim
 
-from agents.Network import Network
-from utils.ReplayBuffer import ReplayBuffer
+from agents.BaseAgent import BaseAgent
 from utils.torch import device, getBatchColumns
 
-class QLearning:
```

Comment:
I moved most of this logic into a `BaseAgent` class so that I could reduce code duplication between the agents. Also this way each agent is much easier to read on its own.